src/libertem/executor/pipelined.py

```diff
@@ -791,9 +900,26 @@ def _run_tasks_inner(
         # of exceptions, it becomes a bit harder to keep attribution of messages to
         # tasks, which is why we have a separate counter for now.
         in_flight = [0]
```

Comment:
I can't recall why this needs to be a list ? Maybe for the use in the closure ? Could possibly get away using:

```python
in_flight = 0
...
def yield_result_if_found(block, timeout):
    nonlocal in_flight
    in_flight -= 1
    ...
...
in_flight += 1
```

Reply:
Yup, that works, too